src/routes/recipes.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { Routes, Route, Navigate } from 'react-router-dom';
+import { useUser } from '../context/userContext';
+import RecipeCategories from '../recipies/RecipeCategories'; 
+import CreateRecipe from '../components/CreateRecipe'; 
+import RecipeList from '../recipies/RecipeList'; 
+import RecipeDetails from '../recipies/RecipeDetails'; 
+import Login from '../components/Login/Login'; 
+
+const ProtectedRoute = ({ element }) => {
+    const { user } = useUser(); 
+    return user ? element : <Navigate to="/login" />; 
+};
+
+const RecipesRoutes = () => {
+    return (
+        <Routes>
+            <Route path="/recipes/categories" element={<RecipeCategories />} />
```

Comment:
Instead of creating separate routes with long path like this, use path groups as we did in the lesson. 
There should be two main groups:
1. recipes
2. user

Under Recipes group, you'll have structure like this:
recipes/
  /create - create recipe page
  /:id - recipe details page
  /categories (a subgroup)
    /:category - specific category page